buyer_data/utils/loan_math.py

```
@@ -0,0 +1,37 @@
+    if rate == 0: return emi * tenure_months
+    power_n = calculate_interest_power(rate, tenure_months)
+    return emi * (power_n - 1) / (rate * power_n)
+
+def calculate_outstanding_balance(principal, rate, emi, months_paid):
+    power_k = calculate_interest_power(rate, months_paid)
+    return (principal * power_k) - (emi * (power_k - 1) / rate)
+
+def calculate_new_tenure_after_prepayment(principal, rate, tenure_months, months_paid, prepayment_amount, emi):
+    balance_k = calculate_outstanding_balance(principal, rate, emi, months_paid)
+    new_principal = balance_k - prepayment_amount
+    if new_principal <= 0: return 0, 0, 0, True
```

Ruff (0.12.2)

Multiple statements on one line (colon)

(E701)